api/grpc_disruption_test.go

```diff
@@ -0,0 +1,377 @@
+						ErrorToReturn:    "ALREADY_EXISTS",
+						OverrideToReturn: "",
+						QueryPercent:     40,
+					},
+				}
+
+				Expect(spec.Validate()).To(BeNil())
+			})
+		})
+
+		Context("which in total have query percentage less than 100", func() {
+			It("Does validate", func() {
+				spec.Endpoints = []v1beta1.EndpointAlteration{
+					{
+						TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+						ErrorToReturn:    "CANCELED",
+						OverrideToReturn: "",
+						QueryPercent:     49,
+					},
+					{
+						TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+						ErrorToReturn:    "ALREADY_EXISTS",
+						OverrideToReturn: "",
+						QueryPercent:     49,
+					},
+				}
+
+				Expect(spec.Validate()).To(BeNil())
+			})
+		})
+
+		Context("which in total have query percentage greater than 100", func() {
+			It("Does not validate", func() {
+				spec.Endpoints = []v1beta1.EndpointAlteration{
+					{
+						TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+						ErrorToReturn:    "CANCELED",
+						OverrideToReturn: "",
+						QueryPercent:     50,
+					},
+					{
+						TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+						ErrorToReturn:    "ALREADY_EXISTS",
+						OverrideToReturn: "",
+						QueryPercent:     51,
+					},
+				}
+
+				Expect(spec.Validate()).ToNot(BeNil())
+			})
+		})
+	})
+
+	Describe("Alterations with ErrorToReturn", func() {
+		Context("which are not in the standard grpc errors", func() {
+			It("Do not validate with ddmark", func() {
+				spec.Endpoints = []v1beta1.EndpointAlteration{
+					{
+						TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+						ErrorToReturn:    "MEOW",
+						OverrideToReturn: "",
+						QueryPercent:     50,
+					},
+				}
+
+				err := ddmark.ValidateStruct(spec, "grpc_test_suite", "github.com/DataDog/chaos-controller/api/v1beta1")
+				Expect(len(err)).To(Equal(1))
+				Expect(err[0].Error()).To(Equal("grpc_test_suite>Endpoints>>ErrorToReturn - ddmark:validation:Enum: field needs to be one of [OK CANCELED UNKNOWN INVALID_ARGUMENT DEADLINE_EXCEEDED NOT_FOUND ALREADY_EXISTS PERMISSION_DENIED RESOURCE_EXHAUSTED FAILED_PRECONDITION ABORTED OUT_OF_RANGE UNIMPLEMENTED INTERNAL UNAVAILABLE DATA_LOSS UNAUTHENTICATED], currently \"MEOW\""))
+			})
+		})
+
+		Context("which are in the standard grpc errors", func() {
+			It("Do validate with ddmarks", func() {
+				spec.Endpoints = []v1beta1.EndpointAlteration{}
+
+				for errorString := range v1beta1.ErrorMap {
+					spec.Endpoints = append(
+						spec.Endpoints,
+						v1beta1.EndpointAlteration{
+							TargetEndpoint:   "/chaos_dogfood.ChaosDogfood/order",
+							ErrorToReturn:    errorString,
+							OverrideToReturn: "",
+							QueryPercent:     1,
+						},
+					)
+				}
+
+				Expect(len(spec.Endpoints)).To(Equal(17))
+
+				err := ddmark.ValidateStruct(spec, "grpc_test_suite", "github.com/DataDog/chaos-controller/api/v1beta1")
+				Expect(len(err)).To(Equal(0))
+			})
+
+			It("Do validate", func() {
```

Comment:
nit: `Do validate` and `do not validate` are a little terse, maybe `passes validation` or `fails validation`?